tests/build.gradle.kts

```diff
@@ -73,78 +73,83 @@ tasks {
-  }
-
+  val testAlwaysSuspending by
+      registering(Test::class) {
+        environment =
+            environment +
+                baseRestateEnvironment(name) +
+                mapOf("RESTATE_WORKER__INVOKER__INACTIVITY_TIMEOUT" to "0s")
+
+        useJUnitPlatform {
+          // Run all the tests with always-suspending or only-always-suspending tag
+          includeTags("always-suspending | only-always-suspending")
+        }
+        // Increase a bit the default timeout
+        systemProperties["junit.jupiter.execution.timeout.testable.method.default"] = "30 s"
+      }
+
+  val testSingleThreadSinglePartition by
+      registering(Test::class) {
+        environment =
+            environment +
+                baseRestateEnvironment(name) +
+                mapOf(
+                    "RESTATE_WORKER__PARTITIONS" to "1",
+                    "RESTATE_TOKIO_RUNTIME__WORKER_THREADS" to "1",
+                    "RESTATE_TOKIO_RUNTIME__MAX_BLOCKING_THREADS" to "1",
+                )
+
+        useJUnitPlatform {
+          // Run all the tests with either no tags, or always-suspending tag
+          includeTags("none() | always-suspending")
+        }
+        // Increase a bit the default timeout
+        systemProperties["junit.jupiter.execution.timeout.testable.method.default"] = "30 s"
+      }
+
+  val testLazyState by
+      registering(Test::class) {
+        environment =
+            environment +
+                baseRestateEnvironment(name) +
+                mapOf(
+                    "RESTATE_WORKER__INVOKER__DISABLE_EAGER_STATE" to "true",
+                )
+
+        useJUnitPlatform {
+          // Run all the tests with either no tags, or always-suspending tag
+          includeTags("lazy-state")
+        }
+      }
+
+  val testPersistedTimers by
+      registering(Test::class) {
+        environment =
+            environment +
+                baseRestateEnvironment(name) +
+                mapOf("RESTATE_WORKER__TIMERS__NUM_TIMERS_IN_MEMORY_LIMIT" to "1")
+
+        useJUnitPlatform {
+          // Run all the tests with always-suspending or only-always-suspending tag
+          includeTags("timers")
+        }
+        // Increase a bit the default timeout
+        systemProperties["junit.jupiter.execution.timeout.testable.method.default"] = "20 s"
+      }
+
+  // Common configuration for all test tasks
   withType<Test>().configureEach {
     dependsOn(":services:http-server:jibDockerBuild")
     dependsOn(":services:java-services:jibDockerBuild")
     dependsOn(":services:node-services:dockerBuild")
 
     maxParallelForks = 3
   }
-}
 
-tasks.named("check") {
-  dependsOn("testAlwaysSuspending")
-  dependsOn("testSingleThreadSinglePartition")
-  dependsOn("testPersistedTimers")
-  dependsOn("testLazyState")
+  check {
+    dependsOn(testAlwaysSuspending)
+    dependsOn(testSingleThreadSinglePartition)
+    dependsOn(testPersistedTimers)
+    dependsOn(testLazyState)
+  }
```

Comment:
Nice, really cool. Now it is actually possible to run individually tests via `--tests`. I think this is now also possible because you've removed the explicit TestReport task. Really good! I guess we can then also close https://github.com/restatedev/e2e/issues/115.